In [1]:
import pandas as pd

In [2]:
df = pd.read_excel("../PI-2/01 - Datasets/Datasets Siniestros Viales/homicidios.xlsx", sheet_name= "HECHOS")
df.head(2)

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE  ... Altura  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...    NaN   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...    NaN   

                                 Cruce  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                  DE LOS CORRALES AV.   

                               Dirección Normalizada COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   

                               XY (CABA)         pos x         pos y  \
0  Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1  Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  

[2 rows x 21 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   AAAA                   696 non-null    int64         
 4   MM                     696 non-null    int64         
 5   DD                     696 non-null    int64         
 6   HORA                   696 non-null    object        
 7   HH                     696 non-null    object        
 8   LUGAR_DEL_HECHO        696 non-null    object        
 9   TIPO_DE_CALLE          696 non-null    object        
 10  Calle                  695 non-null    object        
 11  Altura                 129 non-null    float64       
 12  Cruce                  525 non-null    object        
 13  Direc

Solamente las columnas "Altura" y "Cruce" tienen datos nulos, pero eso se debe a que cuando el siniestro ocurre en un cruce de calle, la altura es dato vacío. La sumatoria de dichos valores NO nulos resulta en 654 valores. 

Por ende, de los 696 registros, hay 42 valores que en ambas columnas se encuentran vacios. A explorar rapidamente esos datos en el dataset, se denota que los mismos indican ocurrencia en AUTOPISTAS.

In [4]:
df = df[["ID", "AAAA", "MM", "HH", "FECHA", "TIPO_DE_CALLE", "Calle", "Cruce", "COMUNA", "pos x", "pos y", "VICTIMA", "ACUSADO"]]
df["FECHA"] = df["FECHA"].dt.day_name()
df.head(2)

ID  AAAA  MM HH     FECHA TIPO_DE_CALLE             Calle  \
0  2016-0001  2016   1  4    Friday       AVENIDA  PIEDRA BUENA AV.   
1  2016-0002  2016   1  1  Saturday      GRAL PAZ    PAZ, GRAL. AV.   

                                 Cruce  COMUNA         pos x         pos y  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.       8  -58.47533969  -34.68757022   
1                  DE LOS CORRALES AV.       9  -58.50877521  -34.66977709   

  VICTIMA    ACUSADO  
0    MOTO       AUTO  
1    AUTO  PASAJEROS

In [5]:
df_homicidios = df.rename(columns={"AAAA":"Año", "MM":"Mes", "HH":"Hora", "FECHA":"Dia", "TIPO_DE_CALLE":"Lugar", "Calle":"Calle_ppal", "Cruce":"Calle_sec", "COMUNA": "Comuna", "pos x": "Longitud", "pos y": "Latitud", "VICTIMA": "Victima", "ACUSADO": "Acusado"})
df_homicidios.head(2)

ID   Año  Mes Hora       Dia     Lugar        Calle_ppal  \
0  2016-0001  2016    1    4    Friday   AVENIDA  PIEDRA BUENA AV.   
1  2016-0002  2016    1    1  Saturday  GRAL PAZ    PAZ, GRAL. AV.   

                             Calle_sec  Comuna      Longitud       Latitud  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.       8  -58.47533969  -34.68757022   
1                  DE LOS CORRALES AV.       9  -58.50877521  -34.66977709   

  Victima    Acusado  
0    MOTO       AUTO  
1    AUTO  PASAJEROS

In [7]:
df_victimas = pd.read_excel("../PI-2/01 - Datasets/Datasets Siniestros Viales/homicidios.xlsx", sheet_name= "VICTIMAS")
df_victimas.head(2)

ID_hecho      FECHA  AAAA  MM  DD        ROL VICTIMA       SEXO EDAD  \
0  2016-0001 2016-01-01  2016   1   1  CONDUCTOR    MOTO  MASCULINO   19   
1  2016-0002 2016-01-02  2016   1   2  CONDUCTOR    AUTO  MASCULINO   70   

   FECHA_FALLECIMIENTO  
0  2016-01-01 00:00:00  
1  2016-01-02 00:00:00

In [8]:
df_victimas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717 entries, 0 to 716
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   ID_hecho             717 non-null    object        
 1   FECHA                717 non-null    datetime64[ns]
 2   AAAA                 717 non-null    int64         
 3   MM                   717 non-null    int64         
 4   DD                   717 non-null    int64         
 5   ROL                  717 non-null    object        
 6   VICTIMA              717 non-null    object        
 7   SEXO                 717 non-null    object        
 8   EDAD                 717 non-null    object        
 9   FECHA_FALLECIMIENTO  717 non-null    object        
dtypes: datetime64[ns](1), int64(3), object(6)
memory usage: 56.1+ KB


In [9]:
df_victimas = df_victimas[["ID_hecho", "ROL", "SEXO", "EDAD", "FECHA_FALLECIMIENTO"]]
df_victimas = df_victimas.rename(columns= {"ID_hecho":"ID", "ROL":"Rol_vic", "SEXO":"Sexo", "EDAD": "Edad", "FECHA_FALLECIMIENTO": "Fecha_fallec" })
df_victimas.head(2)

ID    Rol_vic       Sexo Edad         Fecha_fallec
0  2016-0001  CONDUCTOR  MASCULINO   19  2016-01-01 00:00:00
1  2016-0002  CONDUCTOR  MASCULINO   70  2016-01-02 00:00:00

In [10]:
df_victimas["Sexo"] = df_victimas["Sexo"].replace({"MASCULINO":"M", "FEMENINO":"F"})

In [11]:
# Intenta convertir la columna a fechas
df_victimas["Fecha_fallec"] = pd.to_datetime(df_victimas["Fecha_fallec"], errors = "coerce")
df_victimas.head()

ID    Rol_vic Sexo Edad Fecha_fallec
0  2016-0001  CONDUCTOR    M   19   2016-01-01
1  2016-0002  CONDUCTOR    M   70   2016-01-02
2  2016-0003  CONDUCTOR    M   30   2016-01-03
3  2016-0004  CONDUCTOR    M   18          NaT
4  2016-0005  CONDUCTOR    M   29   2016-02-01

In [12]:
df_victimas["Fecha_fallec"] = df_victimas["Fecha_fallec"].apply(lambda x: "SI" if not pd.isnull(x) else "SD")
df_victimas

ID               Rol_vic Sexo Edad Fecha_fallec
0    2016-0001             CONDUCTOR    M   19           SI
1    2016-0002             CONDUCTOR    M   70           SI
2    2016-0003             CONDUCTOR    M   30           SI
3    2016-0004             CONDUCTOR    M   18           SD
4    2016-0005             CONDUCTOR    M   29           SI
..         ...                   ...  ...  ...          ...
712  2021-0092                PEATON    F   50           SI
713  2021-0093  PASAJERO_ACOMPAÑANTE    F   18           SI
714  2021-0094  PASAJERO_ACOMPAÑANTE    F   43           SI
715  2021-0095             CONDUCTOR    M   27           SI
716  2021-0096             CONDUCTOR    M   60           SI

[717 rows x 5 columns]

In [13]:
# Merge basado en la columna común "ID"
df = pd.merge(df_homicidios, df_victimas, on="ID")
df.head()

ID   Año  Mes Hora       Dia     Lugar        Calle_ppal  \
0  2016-0001  2016    1    4    Friday   AVENIDA  PIEDRA BUENA AV.   
1  2016-0002  2016    1    1  Saturday  GRAL PAZ    PAZ, GRAL. AV.   
2  2016-0003  2016    1    7    Sunday   AVENIDA    ENTRE RIOS AV.   
3  2016-0004  2016    1    0    Sunday   AVENIDA    LARRAZABAL AV.   
4  2016-0005  2016    1    5  Thursday   AVENIDA      SAN JUAN AV.   

                             Calle_sec  Comuna      Longitud       Latitud  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.       8  -58.47533969  -34.68757022   
1                  DE LOS CORRALES AV.       9  -58.50877521  -34.66977709   
2                                  NaN       1  -58.39040293  -34.63189362   
3             VILLEGAS, CONRADO, GRAL.       8  -58.46503904  -34.68092974   
4              SAENZ PE?A, LUIS, PRES.       1  -58.38718297  -34.62246630   

  Victima    Acusado    Rol_vic Sexo Edad Fecha_fallec  
0    MOTO       AUTO  CONDUCTOR    M   19           SI  
1    AUTO  PASAJEROS  CONDUCTOR    M   70           SI  
2    MOTO       AUTO  CONDUCTOR    M   30           SI  
3    MOTO         SD  CONDUCTOR    M   18           SD  
4    MOTO  PASAJEROS  CONDUCTOR    M   29           SI

In [14]:
df_comunas = pd.read_excel("../PI-2/01 - Datasets/Datasets Siniestros Viales/comunas.xlsx")
df_comunas = df_comunas[["COMUNAS", "BARRIOS"]]
df_comunas = df_comunas.rename(columns = {"COMUNAS":"Comuna", "BARRIOS": "Barrios"})
df_comunas.head()

Comuna                                  Barrios
0       2                                 RECOLETA
1       5                          ALMAGRO - BOEDO
2       6                                CABALLITO
3       7                FLORES - PARQUE CHACABUCO
4       9  LINIERS - MATADEROS - PARQUE AVELLANEDA

In [15]:
df_completo = pd.merge(df, df_comunas, on="Comuna")
df_completo = df_completo.drop("ID", axis = 1)
df_completo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 715 entries, 0 to 714
Data columns (total 17 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Año           715 non-null    int64 
 1   Mes           715 non-null    int64 
 2   Hora          715 non-null    object
 3   Dia           715 non-null    object
 4   Lugar         715 non-null    object
 5   Calle_ppal    715 non-null    object
 6   Calle_sec     540 non-null    object
 7   Comuna        715 non-null    int64 
 8   Longitud      715 non-null    object
 9   Latitud       715 non-null    object
 10  Victima       715 non-null    object
 11  Acusado       715 non-null    object
 12  Rol_vic       715 non-null    object
 13  Sexo          715 non-null    object
 14  Edad          715 non-null    object
 15  Fecha_fallec  715 non-null    object
 16  Barrios       715 non-null    object
dtypes: int64(3), object(14)
memory usage: 95.1+ KB


In [17]:
df_completo.to_csv("../PI-2/Datos_Homicidios.csv", index=False)